packages/adaptor-docs/src/DocsPanel.tsx

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import type { PackageDescription } from '@openfn/describe-package';
+import useDocs from './useDocs';
+import Function from './render/Function';
+
+type DocsPanelProps = {
+  specifier?: string;
+  onInsert?: (text: string) => void;
+}
+
+const DocsPanel = ({ specifier, onInsert }: DocsPanelProps) => {
+  if (!specifier) {;
+    return <div>Nothing selected</div>;
+  }
+
+  const pkg = useDocs(specifier);
+  if (pkg === null) {
+    return <div>Loading...</div>
+  }
+  if (pkg === false) {
+    return <div>Failed to load docs.</div>
+  }
+  
+  const { name, version, functions } = pkg as PackageDescription;
+  return (
+    <div className="block m-2">
+      <h1 className="h1 text-lg font-bold text-secondary-700 mb-2">{name} v{version}</h1>
+      <div className="text-sm mb-4">Operations available for use by this adaptor are listed below.</div>
+      {functions
+        // TODO we ought to memo the sort really, although this won't render very often so it's probably ok
+        .sort((a, b) => {
+          if (a.name > b.name) return 1;
+          else if (a.name < b.name) return -1;
+          return 0;
+        })
+        .map((fn) => <Function key={fn.name} fn={fn} onInsert={onInsert} />)}
```

Comment:
I'm getting warnings from React here for duplicate keys, (`each`). In language-common we have `each` and `beta.each` as exports - we are going to have to add the ability to represent exported members with their namespaces.

Reply:
Yep, I clocked that one this morning. Let me raise an issue so it's tracked